pkg/controller/flink/config.go

```diff
@@ -149,11 +152,23 @@ func renderFlinkConfig(app *v1beta1.FlinkApplication) (string, error) {
 	v11, _ := version.NewVersion("1.11")
 
 	if err != nil || appVersion == nil || appVersion.LessThan(v11) {
-		(*config)["jobmanager.heap.size"] = getJobManagerMemory(app)
-		(*config)["taskmanager.heap.size"] = getTaskManagerMemory(app)
+		// if process memory is specified for < 11, error out
+		if app.Spec.JobManagerConfig.SystemMemoryFraction != nil || app.Spec.TaskManagerConfig.SystemMemoryFraction != nil {
+			return "", fmt.Errorf("systemMemoryFraction config cannot be used with flinkVersion < 1.11', use " +
+				"offHeapMemoryFraction instead")
+		}
+
+		(*config)["jobmanager.heap.size"] = getJobManagerHeapMemory(app)
+		(*config)["taskmanager.heap.size"] = getTaskManagerHeapMemory(app)
 	} else {
-		(*config)["jobmanager.memory.process.size"] = getJobManagerMemory(app)
-		(*config)["taskmanager.memory.process.size"] = getTaskManagerMemory(app)
+		// if heap memory is used for >= 1.11, error out
```

Comment:
Not intended to be solved in this PR, but I'm wondering if we can support various flink version --> config compatibility in a different way than `if .. else` blocks. I guess one way could be to couple operator versions to flink versions, but that seems a bit restrictive.

Reply:
I don't think we would want to couple operator versions to flink versions, as most deployments will likely be heterogenous (currently at Lyft we have the operator managing applications with Flink 1.9, 1.10, and 1.11).